### ISSUE TITLE
Add `allow` rule option to `vue/no-template-shadow`

### ISSUE THREAD

**Issue by stefanobartoletti** [vuejs/eslint-plugin-vue#2317]
**What rule do you want to change?**
`vue/no-template-shadow`

**Does this change cause the rule to produce more or fewer warnings?**
Slightly few warnings, in selected occurrences 

**How will the change be implemented? (New option, new default behavior, etc.)?**
Probably the best way is to set a new option to the existing rule, something like this:

```js
"vue/no-template-shadow": ['error', {
    whitelist: ['allowed', 'variable', 'names'],
}],
```

---

I think it can be useful to have the ability to set a whitelist for well known variable names that are allowed to be shadowed, while still being able to throw errors or warnings in all other cases.

It could be possible to have certain selected variable names that have to remain the same (maybe for third party libraries constraints, or for some coding convention), while also wanting to prevent this behavior globally.

**Comment by FloEdelmann** [comment]
To match the `no-shadow` core ESLint rule, we should call that rule property `allow`: https://eslint.org/docs/latest/rules/no-shadow#allow

**Comment by stefanobartoletti** [comment]
Thanks for your reply, I did not even noticed that he core rule already had this feature :D

I'm not so sure if I am able to contribute with a PR, but maybe I can try to have a look in the next few days if I can free some time and maybe share some insights (not sure if I can properly contribute with code)

**Comment by mussinbenarbia** [comment]
@stefanobartoletti @FloEdelmann Hello! I had some free time today and took a stab at it, I hope it's ok 🙏 https://github.com/vuejs/eslint-plugin-vue/pull/2323

**Comment by stefanobartoletti** [comment]
@mussinbenarbia thanks for taking care of this!
